common/src/auth/ServiceConnectionString.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    /// <summary>
+    /// This object handles the connection string for the Azure IoT Services.
+    /// </summary>
+    /// <remarks>
+    /// The connection string contains a set of information that uniquely identify an IoT Service.
+    ///
+    /// A valid connection string shall be in one of the following formats:
+    /// <code>
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessSignature=[Signature]
+    /// </code>
+    ///
+    /// This object parse and store the connection string. It is responsible to provide the authorization token too.
+    /// </remarks>
+    public class ServiceConnectionString
+    {
+        protected static readonly TimeSpan DefaultTokenTimeToLive = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// CONSTRUCOR
+        /// </summary>
+        /// <param name="parser">the <see cref="ServiceConnectionStringParser"/> with the connection string content.</param>
+        /// <exception cref="ArgumentNullException">if the provided parser is null.</exception>
+        public ServiceConnectionString(ServiceConnectionStringParser parser)
+        {
+            if (parser == null)
+            {
+                throw new ArgumentNullException(nameof(parser));
+            }
+
+            HostName = parser.HostName;
+            SharedAccessKeyName = parser.SharedAccessKeyName;
+            SharedAccessKey = parser.SharedAccessKey;
+            SharedAccessSignature = parser.SharedAccessSignatureString;
+            ServiceName = parser.ServiceName;
+            HttpsEndpoint = new UriBuilder("https", parser.HostName).Uri;
+        }
+
+        /// <summary>
+        /// The Service Name
+        /// </summary>
+        public string ServiceName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Client Hostname
+        /// </summary>
+        public string HostName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Client Https Endpoint
+        /// </summary>
+        public Uri HttpsEndpoint
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Audience
+        /// </summary>
+        public string Audience
+        {
+            get { return HostName; }
+        }
+
+        /// <summary>
+        /// The Service Access Policy Name
+        /// </summary>
+        public string SharedAccessKeyName
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// The Service Shared Access Key for the specified
```

Comment:
What is our line length policy? This wraps fairly early.

Reply:
I am not sure if we have one. I will check with team

Reply:
There isn't one at least looking at guideline but for this one I changed to one line